ghcide/src/Development/IDE/Core/Shake.hs

```diff
@@ -1442,7 +1442,7 @@ getAllDiagnostics ::
     STMDiagnosticStore ->
     STM [FileDiagnostic]
 getAllDiagnostics =
-    fmap (concatMap (\(k,v) -> map (fromUri k,ShowDiag,) $ getDiagnosticsFromStore v)) . ListT.toList . STM.listT
+    fmap (concatMap (\(k,v) -> map (\diag -> ideErrorFromLspDiag diag (fromUri k) Nothing) $ getDiagnosticsFromStore v)) . ListT.toList . STM.listT -- TODO: Do we need the structured message here?
```

Comment:
Hmm, this is potentially bad? i.e. do we just throw away the extra info when we put stuff into here? Honestly I have no idea how this thing fits into the diagnostics provided by rules..

Reply:
i.e. I wonder if this is going to bite us when we try to match on the structured error in a handler, maybe it will have been thrown away by this...

Reply:
Fixed in 2cd33dfc